app/models/course/assessment/question/text_response_comprehension_solution.rb

```diff
@@ -32,4 +34,13 @@ def strip_whitespace_solution
   def strip_whitespace_solution_lemma
     solution_lemma.each(&:strip!)
   end
+
+  def validate_solution_lemma_not_empty
+    errors.add(:solution_lemma, :solution_lemma_empty) if solution_lemma.empty?
+  end
+
+  def validate_at_most_one_compre_lifted_word
+    errors.add(:solution_type, :more_than_one_compre_lifted_word) if
+      compre_lifted_word? && point.solutions.count(&:compre_lifted_word?) > 1
```

Comment:
Is it more appropriate to have this validation on `TextResponseComprehensionPoint` instead, since having more than one `compre_lifted_word` Solution renders the Point invalid, but the Solutions themselves are not invalid?

Reply:
Validations should exist on the logical models based on the business logic, so this should be on the `ComprehensionPoint` model. This ensures that when we do changes to business logic we do not miss out changing this piece of code. 

If we want the error message to be more intuitive, one way to do it is that in the `ComprehensionPoint` model validation method, you add the errors to the `solutions` with `compre_lifted_word`.

Reply:
>Validations should exist on the logical models based on the business logic, so this should be on the `ComprehensionPoint` model. This ensures that when we do changes to business logic we do not miss out changing this piece of code.

Okay! I have shifted the multiple `compre_lifted_word` validation to `TextResponseComprehensionPoint` based on the business logic.

>If we want the error message to be more intuitive, one way to do it is that in the `ComprehensionPoint` model validation method, you add the errors to the `solutions` with `compre_lifted_word`.

Decided to do away with this as it will be pretty messy.

Thank you for your inputs!